internal/pkg/cli/app_package.go

```diff
@@ -67,13 +71,23 @@ func newPackageAppOpts(vars packageAppVars) (*packageAppOpts, error) {
 
 	return &packageAppOpts{
 		packageAppVars: vars,
-		ws:             ws,
-		store:          store,
-		describer:      cloudformation.New(sess),
-		runner:         command.New(),
-		stackWriter:    os.Stdout,
-		paramsWriter:   ioutil.Discard,
-		fs:             &afero.Afero{Fs: afero.NewOsFs()},
+		initAddonsSvc: func(o *packageAppOpts) error {
```

Comment:
I wonder if we can hide this in some sort of provider interface. Like, an AddonServiceProvider, which just has one function `ForApp(appName string) (*AddonService, error)`

Just thinking out loud, we don't have to change it now, but that seems like an interested pattern to look into.

Reply:
Yes! Since we have this kind of pattern in the other subcommands. But always open for a better approach for this.

Reply:
I tried the provider pattern but I couldn't get it to work the way I wanted to :( maybe y'all can think of a better way.  I ran into the following problem:

1. The return type for `ForApp` is an `*AddonService`. That means in our tests although we can mock `ForApp` we need to return a concrete type `*AddonService`.
2. Since we're returning a concrete type then you can't mock the call to `Template` 😢

Instead if `ForApp` returned an interface then we could have it return a mock, but returning an interface is usually considered bad practice and I'd rather not go that route.